include/hdf5libs/HDF5RawDataFile.hpp

```diff
@@ -0,0 +1,211 @@
+/**
+ * @file HDF5RawDataFile.hpp
+ *
+ * Class for reading out HDF5 files as
+ *
+ *
+ * This is part of the DUNE DAQ Application Framework, copyright 2020.
+ * Licensing/copyright details are in the COPYING file that you should have
+ * received with this code.
+ */
+
+#ifndef HDF5LIBS_INCLUDE_HDF5LIBS_HDF5RAWDATAFILE_HPP_
+#define HDF5LIBS_INCLUDE_HDF5LIBS_HDF5RAWDATAFILE_HPP_
+
+// DUNE-DAQ
+#include "hdf5libs/hdf5filelayout/Structs.hpp"
+#include "hdf5libs/HDF5FileLayout.hpp"
+
+#include "daqdataformats/Fragment.hpp"
+#include "daqdataformats/TriggerRecord.hpp"
+#include "logging/Logging.hpp"
+
+// External Packages
+#include <highfive/H5DataSet.hpp>
+#include <highfive/H5File.hpp>
+#include <highfive/H5Group.hpp>
+#include <highfive/H5Object.hpp>
+#include <nlohmann/json.hpp>
+
+// System
+#include <filesystem>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <set>
+#include <string>
+#include <sys/statvfs.h>
+#include <variant>
+#include <vector>
+
+namespace dunedaq {
+
+ERS_DECLARE_ISSUE(hdf5libs,
+                  FileOpenFailed,
+                  "Issue when opening file " << file << ": " << message,
+                  ((std::string)file)((std::string)message))
+
+namespace hdf5libs {
+
+/**
+ * @brief HDF5RawDataFile is the class responsible
+ * for interfacing the DAQ format with the HDF5 file format.
+ */
+class HDF5RawDataFile
+{
+
+public:
+  enum
+  {
+    TLVL_BASIC = 2,
+    TLVL_FILE_SIZE = 5
+  };
+
+  // constructor for writing
+  HDF5RawDataFile(std::string file_name,
+                  daqdataformats::run_number_t run_number,
+                  size_t file_index,
+                  std::string application_name,
+                  const hdf5filelayout::FileLayoutParams& fl_params,
+                  unsigned open_flags = HighFive::File::Create);
+  HDF5RawDataFile(std::string file_name,
```

Comment:
This constructors differs from the previous by taking a json configuration in input as opposed to a `hdf5filelayout::FileLayoutParams` struct. It would be preferable to have only one c'tor relying on the config structure.